natlas-server/app/api/utils.py

```diff
@@ -0,0 +1,180 @@
+from flask import current_app
```

Comment:
This file has a lot of unrelated methods. I see scan, screenshots, SSL, etc. This makes it hard to find methods that you're looking for because it ends up in one single file called "utils". What do you think about splitting this file up into different files related to scanning, screenshotting, and different logical groups?

Reply:
We could break it up into a few different groups. The most clear separation is "functions that support /getwork" and "functions that support /submit". I ended up moving them out into utils primarily so that `routes.py` was only routes and any supporting functions weren't in `routes`. I'll break it up a bit further, though, I think you're right.

Reply:
I've currently broken it down into:

`api/prepare_work.py` -> `get_target_tags`, `get_unique_scan_id`, `prepare_work`
`api/process_screenshot.py` -> `create_thumbnail`, `process_screenshots`
`api/process_ssl.py` -> `parse_alt_names`, `parse_subject`, `parse_pubkey`, `parse_ssl_data`
`api/rescan_handler.py` -> `mark_scan_dispatched`, `mark_scan_completed`

I'm wondering if I should create a `processing` folder and then have it more like `api/processing/screenshot.py` and `api/processing/ssl.py` instead. That might be more expandable moving forward, instead of having a bunch of files in the same folder.

Either way, right now each of the above files, with the exception of `rescan_handler.py` really only has to export one function. `prepare_work`, `process_screenshots`, and `parse_ssl_data`. The other functions in each of those files are just used by the exported function.